examples/digital_out/update_port.py

```diff
@@ -0,0 +1,18 @@
+"""Example for generating digital signal.
+
+This example demonstrates how to write values to a digital
+output port.
+"""
+
+import nidaqmx
+from nidaqmx.constants import LineGrouping
+
+with nidaqmx.Task() as task:
+    data = 0xFFFFFFFF
+
+    task.do_channels.add_do_chan("Dev1/port0", line_grouping=LineGrouping.CHAN_FOR_ALL_LINES)
+
+    number_of_samples_written = task.write(data, auto_start=True)
+    print(f"Generate {number_of_samples_written} voltage samples.")
+    task.wait_until_done()
+    task.stop()
```

Comment:
Number of samples and wait until done do not make sense for on-demand.

- It only generates 1 sample.
- task.write() waits until the update is complete

auto_start=True and task.stop() are still relevant though.

Reply:
It's replaced to `while not task.is_task_done()` after `write()` is called.

Reply:
Don't spinwait. Just call stop().

According to https://www.ni.com/docs/en-US/bundle/ni-daqmx/page/whentaskdone.html an on-demand task is not done until you call stop.